orderer/common/filerepo/filerepo_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package filerepo_test
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric/orderer/common/filerepo"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFileRepo_Save(t *testing.T) {
+	options := filerepo.Options{
+		FileSuffix:          ".joinblock",
+		TransientFilePrefix: "~",
+		FileRepoDir:         "testdata",
+	}
+
+	r := filerepo.Repo{
+		Options: options,
+	}
+
+	t.Run("Successful save", func(t *testing.T) {
+		err := r.Save("newchannel", []byte("block-bytes"))
+		defer os.Remove("testdata/newchannel.joinblock")
+		assert.NoError(t, err)
+
+		// Check file bytes
+		bytes, err := ioutil.ReadFile("testdata/newchannel.joinblock")
+		require.NoError(t, err)
+		assert.Equal(t, bytes, []byte("block-bytes"))
```

Comment:
why are we using a mix of `require` and `assert`? can't we use `require` consistently?

Reply:
I saw the slew of PRs recently for standardizing the ledger level tests to use `require` over `assert`. While inherently this doesn't introduce any problems, I feel mildly apprehensive that we may be overusing `require` when `assert` is fine. I think this may come down to what the intended intentions of the `assert` and `require` packages were for initially. I'm fine changing all of these tests to using `require` exclusively though since it doesn't affect anything.

Reply:
`require package is very similar to assert, except for it terminates test if assertion is not successful.` https://goinbigdata.com/testing-go-code-with-testify/ assert was introduced first and then require. 

@btl5037 also mentioned that a similar change happened at the SDK side a few days/weeks back. I do not have a strong opinion on the orderer pkg but my point was that this test uses a mix of both `require` and `assert` which look odd to me.